backend/src/main/java/ai/verta/modeldb/App.java

```diff
@@ -341,4 +325,42 @@ public static void initializeTelemetryBasedOnConfig(MDBConfig mdbConfig)
       LOGGER.info("Telemetry opt out by user");
     }
   }
+
+  @PreDestroy
+  public void onShutDown() {
+    if (App.getInstance().server.isPresent()) {
+      var server = App.getInstance().server.get();
+      try {
+        int activeRequestCount = MonitoringInterceptor.ACTIVE_REQUEST_COUNT.get();
+        while (activeRequestCount > 0) {
+          activeRequestCount = MonitoringInterceptor.ACTIVE_REQUEST_COUNT.get();
+          LOGGER.info("Active Request Count in while:{} ", activeRequestCount);
+          Thread.sleep(1000); // wait for 1s
+        }
+        // Use stderr here since the logger may have been reset by its JVM shutdown
+        // hook.
+        LOGGER.info("*** Shutting down gRPC server since JVM is shutting down ***");
+        server.shutdown();
+        server.awaitTermination();
```

Comment:
Should we call the method with timeout here?

Reply:
Yes, added